packages/core/src/analytics/dispatch.ts

```diff
@@ -0,0 +1,51 @@
+import { CoreContext } from '../context'
+import { CoreSegmentEvent, Callback } from '../events/interfaces'
+import { EventQueue } from '../queue/event-queue'
+import { isOffline } from '../connection'
+import { Emitter } from '../emitter'
+import { invokeCallback } from '../callback'
+
+type DispatchOptions = {
+  timeout?: number
+  debug?: boolean
+  callback?: Callback
+  retryQueue?: boolean
+}
+
+export async function dispatch(
+  event: CoreSegmentEvent,
+  queue: EventQueue,
+  emitter: Emitter,
```

Comment:
This argument is new -- will allow us to decouple this function from listeners like segment inspector

Reply:
Does it make sense for the emitter to ever _not_ be analytics? Wonder if naming the argument something like `analytics` would make it more clear what should be passed/what's emitting events.

Reply:
I prefer the variable name reflect only the minimal required interface, per "principal of least privelege / interface segration principal", rather than the larger concrete class -- this is  at minimum helpful for unit tests =)

Reply:
To be clear - I was expecting the type to still be `Emitter`, just the name of the property to be `analytics`. Can we have a doc string then so someone looking at the function understands we expect it to be analytics (even if unit tests will pass anything)

Reply:
I prefer the name to reflect the type, since if you pass an emitter that's not analytics, it's a bit confusing.